tendermint-rs/src/merkle.rs

```diff
@@ -0,0 +1,116 @@
+//! Merkle tree used in Tendermint networks
+
+use sha2::{Digest, Sha256};
+
+/// Size of Merkle root hash
+pub const HASH_SIZE: usize = 32;
+
+/// Compute a simple Merkle root from the arbitrary sized byte slices
+pub fn simple_hash_from_byte_slices(byte_slices: &[&[u8]]) -> [u8; HASH_SIZE] {
+    let length = byte_slices.len();
```

Comment:
Now I notice it'd need to be an [ExactSizeIterator](https://doc.rust-lang.org/std/iter/trait.ExactSizeIterator.html) heh, so slightly less trivial...